In [1]:
import pandas as pd

### 최종 모델 데이터 가져옴 

In [3]:
df = pd.read_csv('data/모델돌릴꺼2.csv')
df.head(2)

Unnamed: 0      라인   win  분당 킬  분당 어시  분당 데스   퍼블유무  퍼블어시유무  분당 얻은 골드  \
0           0     TOP  True  0.54   0.11   0.22  False   False    509.12   
1           1  JUNGLE  True  0.14   0.69   0.14  False    True    382.95   

   분당 피해량  분당 cc건시간  분당 CS  분당 죽은시간  분당 시야점수  
0  862.20      0.98    5.0     5.65     0.36  
1  570.17      0.80    1.0     2.97     0.76

### Unnamed: 0 제거

In [4]:
df.drop('Unnamed: 0',1,inplace=True)
df.head(2)

C:\Users\sujong\AppData\Local\Temp\ipykernel_6188\3130116340.py:1: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  df.drop('Unnamed: 0',1,inplace=True)


라인   win  분당 킬  분당 어시  분당 데스   퍼블유무  퍼블어시유무  분당 얻은 골드  분당 피해량  \
0     TOP  True  0.54   0.11   0.22  False   False    509.12  862.20   
1  JUNGLE  True  0.14   0.69   0.14  False    True    382.95  570.17   

   분당 cc건시간  분당 CS  분당 죽은시간  분당 시야점수  
0      0.98    5.0     5.65     0.36  
1      0.80    1.0     2.97     0.76

In [5]:
df['라인'].unique()

array(['TOP', 'JUNGLE', 'MIDDLE', 'BOTTOM', 'UTILITY'], dtype=object)

### 각 라인별로 데이터 나누기

In [6]:
top = df.loc[df['라인']=='TOP']
jungle = df.loc[df['라인']=='JUNGLE']
mid = df.loc[df['라인']=='MIDDLE']
bottom = df.loc[df['라인']=='BOTTOM']
utility = df.loc[df['라인']=='UTILITY']

### 라인 구분할때 사용했던 컬럼 제거

In [7]:
top.drop(['라인'],1, inplace=True)
jungle.drop(['라인'],1, inplace=True)
mid.drop(['라인'],1, inplace=True)
bottom.drop(['라인'],1, inplace=True)
utility.drop(['라인'],1, inplace=True)

C:\Users\sujong\AppData\Local\Temp\ipykernel_6188\2549572576.py:1: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  top.drop(['라인'],1, inplace=True)
C:\Users\sujong\AppData\Local\Temp\ipykernel_6188\2549572576.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  top.drop(['라인'],1, inplace=True)
C:\Users\sujong\AppData\Local\Temp\ipykernel_6188\2549572576.py:2: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only.
  jungle.drop(['라인'],1, inplace=True)
C:\Users\sujong\AppData\Local\Temp\ipykernel_6188\2549572576.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation:

### 탑의 승리와 가장 연관있는 지표 알아보기

In [8]:
top.corr()

win      분당 킬     분당 어시     분당 데스      퍼블유무    퍼블어시유무  \
win       1.000000  0.338310  0.336567 -0.442457  0.067262  0.050431   
분당 킬      0.338310  1.000000  0.140860 -0.176069  0.252319 -0.003316   
분당 어시     0.336567  0.140860  1.000000 -0.082527 -0.002509  0.112738   
분당 데스    -0.442457 -0.176069 -0.082527  1.000000 -0.026480 -0.049485   
퍼블유무      0.067262  0.252319 -0.002509 -0.026480  1.000000 -0.086728   
퍼블어시유무    0.050431 -0.003316  0.112738 -0.049485 -0.086728  1.000000   
분당 얻은 골드  0.444460  0.827638  0.168239 -0.303028  0.214478  0.023820   
분당 피해량    0.201623  0.709484  0.294190  0.028736  0.156963  0.009289   
분당 cc건시간  0.046946  0.149437  0.277897  0.001949  0.026921  0.027493   
분당 CS     0.129293  0.201767 -0.198751 -0.285629  0.053776  0.005091   
분당 죽은시간  -0.358863 -0.060889  0.015251  0.870565 -0.004600 -0.041852   
분당 시야점수   0.139445  0.125854  0.174969 -0.107345  0.032727  0.023640   

          분당 얻은 골드    분당 피해량  분당 cc건시간     분당 CS   분당 죽은시간   분당 시야점수  
win       0.444460  0.201623  0.046946  0.129293 -0.358863  0.139445  
분당 킬      0.827638  0.709484  0.149437  0.201767 -0.060889  0.125854  
분당 어시     0.168239  0.294190  0.277897 -0.198751  0.015251  0.174969  
분당 데스    -0.303028  0.028736  0.001949 -0.285629  0.870565 -0.107345  
퍼블유무      0.214478  0.156963  0.026921  0.053776 -0.004600  0.032727  
퍼블어시유무    0.023820  0.009289  0.027493  0.005091 -0.041852  0.023640  
분당 얻은 골드  1.000000  0.670014  0.063888  0.556252 -0.158799  0.200020  
분당 피해량    0.670014  1.000000  0.234509  0.200173  0.216587  0.185702  
분당 cc건시간  0.063888  0.234509  1.000000 -0.111211  0.031006  0.069441  
분당 CS     0.556252  0.200173 -0.111211  1.000000 -0.243296  0.120512  
분당 죽은시간  -0.158799  0.216587  0.031006 -0.243296  1.000000 -0.014304  
분당 시야점수   0.200020  0.185702  0.069441  0.120512 -0.014304  1.000000

탑의 경우 관련있는 상위 5개 항목   
- 분당 얻은 골드 0.444
- 분당 데스 -0.442
- 분당 죽은시간 -0.359
- 분당 킬 0.338
- 분당 어시 0.337

  
탑의 경우 관련있는 하위 5개 항목  
- 분당 cc건 시간 0.0469
- 퍼블 어시유무 0.0504
- 퍼블 유무 0.0673
- 분당 cs 0.129
- 분당 시야점수 0.139 

### 정글의 승리와 가장 연관있는 지표 알아보기

In [9]:
jungle.corr()

win      분당 킬     분당 어시     분당 데스      퍼블유무    퍼블어시유무  \
win       1.000000  0.329481  0.372398 -0.524755  0.053865  0.063986   
분당 킬      0.329481  1.000000  0.091682 -0.173425  0.155815  0.014106   
분당 어시     0.372398  0.091682  1.000000 -0.135044  0.005384  0.154181   
분당 데스    -0.524755 -0.173425 -0.135044  1.000000 -0.048833 -0.035949   
퍼블유무      0.053865  0.155815  0.005384 -0.048833  1.000000 -0.106002   
퍼블어시유무    0.063986  0.014106  0.154181 -0.035949 -0.106002  1.000000   
분당 얻은 골드  0.472286  0.865487  0.237797 -0.335931  0.140766  0.043565   
분당 피해량    0.189947  0.734447  0.285775  0.046231  0.095531  0.028472   
분당 cc건시간  0.033904  0.059185  0.220537 -0.003722  0.010886  0.018877   
분당 CS    -0.008600  0.196449  0.007843  0.007316  0.022881  0.008425   
분당 죽은시간  -0.442531 -0.073637 -0.058266  0.879009 -0.029615 -0.029916   
분당 시야점수   0.168032  0.102556  0.175515 -0.134933  0.037256  0.050242   

          분당 얻은 골드    분당 피해량  분당 cc건시간     분당 CS   분당 죽은시간   분당 시야점수  
win       0.472286  0.189947  0.033904 -0.008600 -0.442531  0.168032  
분당 킬      0.865487  0.734447  0.059185  0.196449 -0.073637  0.102556  
분당 어시     0.237797  0.285775  0.220537  0.007843 -0.058266  0.175515  
분당 데스    -0.335931  0.046231 -0.003722  0.007316  0.879009 -0.134933  
퍼블유무      0.140766  0.095531  0.010886  0.022881 -0.029615  0.037256  
퍼블어시유무    0.043565  0.028472  0.018877  0.008425 -0.029916  0.050242  
분당 얻은 골드  1.000000  0.703299  0.069750  0.324348 -0.205053  0.165466  
분당 피해량    0.703299  1.000000  0.111277  0.321852  0.218552  0.137090  
분당 cc건시간  0.069750  0.111277  1.000000  0.021512  0.027226  0.062897  
분당 CS     0.324348  0.321852  0.021512  1.000000  0.084029  0.050221  
분당 죽은시간  -0.205053  0.218552  0.027226  0.084029  1.000000 -0.081619  
분당 시야점수   0.165466  0.137090  0.062897  0.050221 -0.081619  1.000000

정글의 경우 관련있는 상위 5개 항목   
- 포탑어시 0.57
- kda 0.498    
- 시간대비 죽은시간 -0.44  
- 데스 -0.434  
- 연속킬 0.36  
  
정글의 경우 관련있는 하위 5개 항목 
- cs수 -0.00753   
- 설치 와드수 0.000796  
- 핑와 갯수 0.0236  
- 와드 제거수 -0.0305  
- cc건 시간 0.0312    

### 미드의 승리와 가장 연관있는 지표 알아보기

In [ ]:
mid.corr()

win        킬수        어시        데스  챔피언에게 가한 피해량  \
win               1.000000  0.293509  0.305253 -0.366544      0.156541   
킬수                0.293509  1.000000  0.264348  0.108975      0.767053   
어시                0.305253  0.264348  1.000000  0.183097      0.524197   
데스               -0.366544  0.108975  0.183097  1.000000      0.325533   
챔피언에게 가한 피해량      0.156541  0.767053  0.524197  0.325533      1.000000   
받은 피해량           -0.090832  0.452658  0.383119  0.655027      0.639379   
얻은 골드             0.242871  0.781286  0.497153  0.287097      0.863857   
cs 수              0.065190  0.344931  0.277535  0.136671      0.570590   
포탑 골드 획득 갯수       0.611636  0.372316  0.362438 -0.177800      0.343512   
퍼블유무              0.059787  0.173192 -0.014456 -0.004397      0.088726   
퍼블어시유무            0.054920  0.000129  0.109661 -0.028837      0.010700   
연속 킬              0.362311  0.774823  0.164614 -0.247098      0.501828   
시야점수              0.124125  0.296810  0.416062  0.187104      0.476178   
핑와산갯수             0.029985  0.053613  0.125813  0.004017      0.090016   
와드제거수             0.074762  0.191130  0.215375  0.091034      0.264134   
설치와드수             0.047767  0.215491  0.401015  0.243724      0.430931   
게임시간 대비 cc기 건 시간  0.050738  0.029466  0.273629 -0.004039      0.167474   
게임시간 대비 죽은 시간    -0.370563  0.086783  0.102023  0.940822      0.264423   
kda               0.465101  0.366079  0.290266 -0.558134      0.205129   

                    받은 피해량     얻은 골드      cs 수  포탑 골드 획득 갯수      퍼블유무  \
win              -0.090832  0.242871  0.065190     0.611636  0.059787   
킬수                0.452658  0.781286  0.344931     0.372316  0.173192   
어시                0.383119  0.497153  0.277535     0.362438 -0.014456   
데스                0.655027  0.287097  0.136671    -0.177800 -0.004397   
챔피언에게 가한 피해량      0.639379  0.863857  0.570590     0.343512  0.088726   
받은 피해량            1.000000  0.681867  0.463686     0.119280  0.044526   
얻은 골드             0.681867  1.000000  0.757862     0.476047  0.088107   
cs 수              0.463686  0.757862  1.000000     0.324593 -0.005243   
포탑 골드 획득 갯수       0.119280  0.476047  0.324593     1.000000  0.057899   
퍼블유무              0.044526  0.088107 -0.005243     0.057899  1.000000   
퍼블어시유무            0.002323  0.016725  0.001701     0.052162 -0.088597   
연속 킬              0.157519  0.548949  0.263425     0.385106  0.120527   
시야점수              0.403054  0.558360  0.486903     0.252661  0.013263   
핑와산갯수             0.064169  0.130074  0.155148     0.053538  0.008733   
와드제거수             0.238484  0.340712  0.326538     0.134051  0.016662   
설치와드수             0.387666  0.491783  0.449817     0.199626 -0.013871   
게임시간 대비 cc기 건 시간  0.011074  0.088294  0.078295     0.104606 -0.006699   
게임시간 대비 죽은 시간     0.579289  0.213804  0.088463    -0.198977  0.004025   
kda              -0.156064  0.250746  0.125350     0.412697  0.069866   

                    퍼블어시유무      연속 킬      시야점수     핑와산갯수     와드제거수     설치와드수  \
win               0.054920  0.362311  0.124125  0.029985  0.074762  0.047767   
킬수                0.000129  0.774823  0.296810  0.053613  0.191130  0.215491   
어시                0.109661  0.164614  0.416062  0.125813  0.215375  0.401015   
데스               -0.028837 -0.247098  0.187104  0.004017  0.091034  0.243724   
챔피언에게 가한 피해량      0.010700  0.501828  0.476178  0.090016  0.264134  0.430931   
받은 피해량            0.002323  0.157519  0.403054  0.064169  0.238484  0.387666   
얻은 골드             0.016725  0.548949  0.558360  0.130074  0.340712  0.491783   
cs 수              0.001701  0.263425  0.486903  0.155148  0.326538  0.449817   
포탑 골드 획득 갯수       0.052162  0.385106  0.252661  0.053538  0.134051  0.199626   
퍼블유무             -0.088597  0.120527  0.013263  0.008733  0.016662 -0.013871   
퍼블어시유무            1.000000  0.013702  0.024371  0.008048  0.019856  0.013047   
연속 킬              0.013702  1.000000  0.193228  0.050614  0.132349  0.111698   
시야점수              0.024371  0.193228  1.0